atomic_reactor/plugins/build_imagebuilder.py

```diff
@@ -0,0 +1,77 @@
+"""
+Copyright (c) 2018 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+import subprocess
+from time import sleep
+from six import PY2
+
+from atomic_reactor.plugin import BuildStepPlugin
+from atomic_reactor.build import BuildResult
+from atomic_reactor.constants import CONTAINER_IMAGEBUILDER_BUILD_METHOD
+
+
+def sixdecode(data):
+    return data.decode() if PY2 else data
+
+
+class ImagebuilderPlugin(BuildStepPlugin):
+    """
+    Build image using imagebuilder https://github.com/openshift/imagebuilder
+    This requires the imagebuilder executable binary to be in $PATH.
+    """
+
+    key = CONTAINER_IMAGEBUILDER_BUILD_METHOD
+
+    def run(self):
+        """
+        Build image inside current environment using imagebuilder;
+        It's expected this may run within (privileged) docker container.
+
+        Returns:
+            BuildResult
+        """
+        builder = self.workflow.builder
+
+        image = builder.image.to_str()
+        # TODO: directly invoke go imagebuilder library in shared object via python module
+        kwargs = dict(stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if not PY2:
+            kwargs['encoding'] = 'utf-8'
+        ib_process = subprocess.Popen(['imagebuilder', '-t', image, builder.df_dir], **kwargs)
+
+        self.log.debug('imagebuilder build has begun; waiting for it to finish')
+        (output, errors) = ([], [])
+        while True:
+            poll = ib_process.poll()
+            out = sixdecode(ib_process.stdout.readline())
+            if out:
+                self.log.info(out.strip())
+                output.append(out)
+            err = sixdecode(ib_process.stderr.readline())
+            if err:
+                self.log.error(err.strip())
+                output.append(err)  # include stderr with stdout
+                errors.append(err)  # as well as separately
+            if out == '' and err == '':
+                if poll is not None:
+                    break
+                sleep(0.1)  # don't busy-wait when there's no output
```

Comment:
This line doesn't get covered during unit tests. Is it feasible to arrange for tests to hit this line? For example, maybe a test could `flexmock(time).should_receive('sleep').with_args(0.1).and_raise(NotImplementedError)`, and then run the plugin `with pytest.raises(NotImplementedError): ...`.